src/sdk/pynni/nni/nas/tensorflow/mutables.py

```diff
@@ -94,21 +92,18 @@ def __init__(self, op_candidates, reduction='sum', return_mask=False, key=None):
         self.choices = op_candidates
         self.reduction = reduction
         self.return_mask = return_mask
-        self._built = False
 
     def call(self, *inputs):
-        if not self._built:
-            for op in self.choices:
-                if len(inputs) > 1:  # FIXME: not tested
-                    op.build([inp.shape for inp in inputs])
-                elif len(inputs) == 1:
-                    op.build(inputs[0].shape)
-            self._built = True
         out, mask = self.mutator.on_forward_layer_choice(self, *inputs)
         if self.return_mask:
             return out, mask
         return out
 
+    def build(self, input_shape):
+        self._check_built()
+        for op in self.choices:
+            op.build(input_shape)
```

Comment:
what is the meaning of `op.build`?

Reply:
TF layers do not require input shape to `__init__`, so they create weights when `build` is called.
Normally this happens implicitly at first forward pass. We trigger it manually because only one (or some) candidate layer is connected at first forward so other candidate layers cannot inference their input shape automatically.

Reply:
got it, thanks.